internal/internal_worker.go

```diff
@@ -1668,6 +1671,7 @@ func NewAggregatedWorker(client *WorkflowClient, taskQueue string, options Worke
 		WorkerBuildID:                         options.BuildID,
 		UseBuildIDForVersioning:               options.UseBuildIDForVersioning,
 		DeploymentName:                        options.DeploymentName,
+		DefaultVersioningBehavior:             options.DefaultVersioningBehavior,
```

Comment:
As also discussed offline this morning, I think it's better to wrap these options in a struct called `DeploymentOption` or something. The list may grow in the future and we don't want the user to have to repeat all the fields individually in each TQ. Example:
```
deploymentOptions := worker.DeploymentOptions{
	DeploymentName: "my-app",
	BuildID: os.Getenv("BUILD_ID"),
	UseBuildIDForVersioning: true,
	DefaultVersioningBehavior: workflow.VersioningBehaviorAutoUpgrade,
}

w1 := worker.NewWorker(client, "foo", worker.WorkerOptions{
	DeploymentOptions: deploymentOptions,
})

w2 := worker.NewWorker(client, "bar", worker.WorkerOptions{
	DeploymentOptions: deploymentOptions,
})
```

Reply:
I'm happy with this. @cretz and @Quinn-With-Two-Ns shall I add the struct and deprecate the original BuildID and UseBuildIdForVersioning?

Reply:
All these fields are marked experimental right? If so I have no strong opinion here. If we do this in Go we should probably group them in all SDKs though.